onchaind/onchain.c

```diff
@@ -2117,6 +2117,94 @@ static void handle_their_unilateral(const struct bitcoin_tx *tx,
 	wait_for_resolved(outs);
 }
 
+static void handle_unknown_commitment(const struct bitcoin_tx *tx,
+				      u32 tx_blockheight,
+				      u64 commit_num,
+				      const struct bitcoin_txid *txid,
+				      const struct pubkey *possible_remote_per_commitment_point,
+				      const struct basepoints basepoints[NUM_SIDES],
+				      const struct htlc_stub *htlcs,
+				      const bool *tell_if_missing,
+				      struct tracked_output **outs)
+{
+	struct keyset *ks;
+	int to_us_output = -1;
+	u8 *local_script;
+
+	resolved_by_other(outs[0], txid, UNKNOWN_UNILATERAL);
+
+	if (!possible_remote_per_commitment_point)
+		goto search_done;
+
+	keyset = ks = tal(tx, struct keyset);
+	if (!derive_keyset(possible_remote_per_commitment_point,
+			   &basepoints[REMOTE],
+			   &basepoints[LOCAL],
+			   ks))
+		status_failed(STATUS_FAIL_INTERNAL_ERROR,
+			      "Deriving keyset for possible_remote_per_commitment_point %s",
+			      type_to_string(tmpctx, struct pubkey,
+					     possible_remote_per_commitment_point));
+
+	local_script = scriptpubkey_p2wpkh(tmpctx, &keyset->other_payment_key);
+	for (size_t i = 0; i < tal_count(tx->output); i++) {
+		struct tracked_output *out;
+
+		if (local_script
+		    && scripteq(tx->output[i].script, local_script)) {
+			/* BOLT #5:
+			 *
+			 * - MAY take no action in regard to the associated
+			 *   `to_remote`, which is simply a P2WPKH output to
+			 *   the *local node*.
+			 *   - Note: `to_remote` is considered *resolved* by the
+			 *     commitment transaction itself.
+			 */
+			out = new_tracked_output(&outs, txid, tx_blockheight,
+						 UNKNOWN_UNILATERAL,
+						 i, tx->output[i].amount,
+						 OUTPUT_TO_US, NULL, NULL, NULL);
+			ignore_output(out);
+			local_script = NULL;
+
+			/* Tell the master that it will want to add
+			 * this UTXO to its outputs */
+			wire_sync_write(REQ_FD, towire_onchain_add_utxo(
+						    tmpctx, txid, i,
+						    possible_remote_per_commitment_point,
+						    tx->output[i].amount,
+						    tx_blockheight));
+			to_us_output = i;
+			continue;
```

Comment:
This is pretty much a no-op, can be removed?

Reply:
Agreed.  Cut & paste from the normal unilateral case, which has other branches....